server/src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -359,6 +359,9 @@ public ActionRequestValidationException validate() {
             }
         }
         if (source != null) {
+            if (source.query() != null && source.queries().isEmpty() == false) {
```

Comment:
I have it set up to do linear combination by default since that's the same behavior for multiple knn queries now. Once things like sort and rescore are possibly added we would have to know we can't just shortcut them into a boolean query anymore.

Reply:
> Once things like sort and rescore are possibly added we would have to know we can't just shortcut them into a boolean query anymore.

If this is a future concern, we should disallow it now instead of switching behavior later.

Reply:
I apologize as I didn't explain all of my thoughts on this. I think what is in this PR now is the behavior that makes the most sense since we already default to linear combination for some combination of query and knn searches. What we can do in the future is see if queries is just a set of *only* QueryBuilders with no sorts, no re-scores, etc. and continue this default behavior, otherwise we can use an (interal?) linear combination ranker to execute each query separately like we do for RRF now. This should only be a bit of extra logic as part of `if/else` block.

Reply:
After more thought, I will go ahead and disallow this as both Ben and Luca suggest. I still think that it's more consistent to default to linear combination, but we can always add it later.